### ISSUE TITLE
Variable-length chunks

### ISSUE THREAD

**Issue by TomNicholas** [earth-mover/icechunk#386]
I would like to be able to use zarr + virtualizarr + icechunk with variable-length chunks - see 
https://github.com/zarr-developers/zarr-specs/issues/138.

I'm thinking about what changes in the stack would be required to get this to work - definitely in zarr-python, but also presumably icechunk needs to be able to return chunks of variable size, and the icechunk spec has to accommodate that generality? Then if it's in the icechunk spec does it also need to be in the zarr spec too?

**Comment by TomNicholas** [comment]
> Does the solution need to scale to accommodate arbitrarily large arrays?

Yeah this is the key question, from which everything else should follow.

> In summary, what I think is needed to move forward is a set of meticulously documented use cases and technical requirements.

I'll start: There's one here that's representative of the output of many HPC fluid simulation codes: https://github.com/zarr-developers/VirtualiZarr/issues/217

**Comment by dcherian** [comment]
Another extremely common one is virtual references to netCDF files, which have daily frequency data and a single month of data in a single netCDF file.

**Comment by abarciauskas-bgse** [comment]
@dcherian I've heard this example before but not sure what datasets are structured this way - can you provide an example?

**Comment by rabernat** [comment]
> can you provide an example?

Most of these files fit the bill: https://nsf-ncar-era5.s3.amazonaws.com/index.html#e5.oper.an.sfc/ 

The files are partitioned by month and chunked in such a way that the time-concatenated chunks would be uneven.
